v2/client/invoker.go

```diff
@@ -39,50 +40,55 @@ type receiveInvoker struct {
 func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn protocol.ResponseFn) (err error) {
 	var isFinished bool
 	defer func() {
+		// skip finishing if we did it explicitly already
 		if !isFinished {
-			if err2 := m.Finish(err); err2 == nil {
-				err = err2
+			if fErr := m.Finish(err); fErr == nil {
```

Comment:
ah well there is a logic bug in the original code... if `err2 == nil` should be `err2 != nil`

This will prevent any error from m.Finish to be returned. I like the idea that m.Finish can eat the error. 

Can you return fErr if it is not nil?

maybe it is as simple as changing this whole block to:

```err = m.Finish(err)```

Reply:
Done